buildbot/buildbot/worker.py

```diff
@@ -235,15 +235,13 @@ def stage_after_make(task):
             ))
 
         if config['TOOLCHAIN'] == 'f1':
-            # Get the AWS platform ID for F1 builds.
-            platform_script = (
-                'cd $AWS_FPGA_REPO_DIR ; '
-                'source ./sdaccel_setup.sh > /dev/null ; '
-                'echo $AWS_PLATFORM'
+            # Configure current environment.
+            source_script = (
+                'source $AWS_FPGA_REPO_DIR/sdaccel_setup.sh ; '
             )
-            proc = task.run([platform_script], capture=True, shell=True)
-            aws_platform = proc.stdout.decode('utf8').strip()
+            proc = task.run([source_script], capture=True, shell=True)
 
+            aws_platform = os.environ('AWS_PLATFORM')
```

Comment:
Sadly, I don't this will quite work. `os.environ` is the environment of the _current_ process, i.e., the buildbot worker. We need the variable from the _subprocess_, i.e., the shell that just ran the `source` command. That's the reason for the in-shell `echo`.

Reply:
Ah, thanks for the explanation! Reverted.